src/express/server.js

```diff
@@ -160,32 +163,48 @@ class ExpressServer {
       authMiddleware(namespace, options),
       dataLoaderMiddleware(namespace),
       renderHTMLMiddleware(namespace, webpackAssets, loadableStats, options),
-      (req, res) => {
-        const statusOK = 200
-        const statusRedirect = 301
-        const { html, routerStaticContext } = req[namespace]
-
+      function handleRedirect(req, res, next) {
+        const { routerStaticContext } = req[namespace]
         if (routerStaticContext.url) {
           // somewhere a `<Redirect>` was rendered
-          res.redirect(statusRedirect, routerStaticContext.url)
+          res.redirect(
+            statusCodeConst.movedPermanently,
+            routerStaticContext.url
+          )
           return
         }
 
-        const statusCode = _.get(routerStaticContext, 'statusCode', statusOK)
-        if (!res.headersSent && statusCode < statusRedirect) {
+        const statusCode = _.get(
+          routerStaticContext,
+          'statusCode',
+          statusCodeConst.ok
+        )
+        res.locals.statusCode = statusCode
+        next()
+      },
+      function setResponseHeader(req, res, next) {
+        if (globalEnv.releaseBranch === releaseBranchConsts.preview) {
```

Comment:
Leave the comment here to describe why we disable cache on preview branch

Reply:
Oops. Forget to add comments.